sigrok and friends: Update to nightly

vifino opened the pull request:
## Description of changes

The Sigrok project hasn't had a release in a long time and there is no timeline on when that happens.

However, development is far from stopped.
On the project's site, it is recommended to use the nightly builds.
I asked on IRC about it and it was recommended to package nightly releases until the project publishes releases again.

This advances the packaged version roughly two years or more.
All in one PR, as it doesn't make sense to update one or the other.

In addition, I packaged smuview, which is also "a sigrok project", even though it doesn't use the sigrok.org Git.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x]  Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

vifino committed:
libsigrok: 0.5.2 -> 0.5.2-unstable-2024-01-03

vifino committed:
libsigrokdecode: 0.5.3 -> 0.5.3-unstable-2023-10-23

vifino committed:
sigrok-cli: 0.7.2 -> 0.7.2-unstable-2023-04-10

vifino committed:
pulseview: 0.4.2 -> 0.4.2-unstable-2024-01-26

vifino committed:
smuview: init at 0.0.5-unstable-2023-04-12

panicgh commented on pkgs/applications/science/electronics/pulseview/default.nix:
```diff
@@ -1,45 +1,38 @@
 stdenv.mkDerivation rec {
   pname = "pulseview";
-  version = "0.4.2";
+  version = "nightly-2023-06-16";
```
Looking through nixpkgs, `"unstable-YYYY-MM-DD"` is much more common than `"nighly-YYYY-MM-DD"`.

```suggestion
  version = "unstable-2023-06-16";
```

(also for the other files and commit msgs)

vifino replied:
Oh, yes, that is true.

However, the project themselves calls it nightly and there was a lot of heated discussion about the nightlies being called "unstable", I think some people might feel attacked.
This is what a project member wanted it to be called.

So, IMO, better to stick with nightly.

panicgh commented on pkgs/development/tools/libsigrok/default.nix:
```diff
@@ -1,6 +1,7 @@
 { lib
```
Changes in this file are part of the sigrok-firmware-fx2lafw commit (e98efa6742cdfd531b14a2aa2323389a83770d5e) but belong in the libsigrok (dc21884dfaa971b6b5a3f5f74bd746e38166bb20) commit, which is almost empty at the moment.

vifino replied:
Argh! Guess I need to dig out the git skills to fix this.
Thanks for noticing!

panicgh reviewed (commented):
I've also asked for any release plans a few weeks ago and received the same answer as you did. Then I also updated the packages locally to their unstable versions, but I never dared to upstream them. I'll compare what I did with your proposal and add more suggestions where useful.
In general, I agree that we should ship unstable packages for now as upstream hasn't made releases for so long.

vifino committed:
sigrok-firmware-fx2lafw: 0.1.7 -> 0.1.7-unstable-2024-02-03

panicgh commented on pkgs/applications/science/electronics/smuview/default.nix:
```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, libtool
+, pkg-config
+, cmake
+, glib
+, boost
+, libsigrok
+, libserialport
+, libzip
+, libftdi1
+, hidapi
+, glibmm
+, python3
+, qtsvg
+, qwt
+, qttools
```
smuview builds for me without libtool, qtsvg, qttools.

vifino replied:
I looked into the [building guide](https://sigrok.org/wiki/Building#SmuView) for the dependencies, but I guess they might be outdated as well.

I'll remove them.

bjornfor commented on pkgs/development/tools/libsigrok/default.nix:
```diff
@@ -19,16 +20,17 @@
 stdenv.mkDerivation rec {
   pname = "libsigrok";
-  version = "0.5.2";
+  version = "nightly-2023-10-05";
```
IMHO, committer date makes most sense.

vifino replied:
I concur.

I chose the date shown by `git log`, which is author date,
but I didn't think about it.

I'll update all the packages to the current committer date.

vifino commented on pkgs/applications/science/electronics/pulseview/default.nix:
```diff
@@ -1,45 +1,38 @@
 
 stdenv.mkDerivation rec {
   pname = "pulseview";
-  version = "0.4.2";
+  version = "nightly-2023-06-16";
```
Ah, a typo! Thank you for paying this much attention to my PR.
I really appreciate it. :)

vifino commented on pkgs/applications/science/electronics/smuview/default.nix:
```diff
@@ -0,0 +1,69 @@
+  ];
+
+  meta = with lib; {
+    description = "SmuView is a GUI for sigrok that supports power supplies, electronic loads and all sorts of measurement devices like multimeters, LCR meters and so on";
```
I haven't used `longDescription` yet, TIL, thank you!

vifino commented on pkgs/applications/science/electronics/pulseview/default.nix:
```diff
@@ -1,45 +1,38 @@
-    })
+  ] ++ lib.optionals stdenv.isLinux [
+    bluez
+    util-linux
+    libselinux
+    libsepol
```
I'm not convinced either, but I am not sure if having warnings about missing libraries is better, especially if it ends up affecting behavior, which I am not certain about.

Perhaps it is worth looking into fixing the `gio` pkgconf file if they are truly unused?
If they'd be used and not dead code, the linker would emit an unresolved symbol error, right?

vifino commented:
Hey @panicgh, thank you kindly for your review(s)!

I made updated all the versions to what `git show -s --format=nightly-%cs` showed me and removed the unused dependencies in my smuview derivation you mentioned .

I am still unsure about the `util-linux` and friends, but I'll remove them as well, after all, the linker doesn't complain about missing symbols.

panicgh reviewed (approved):
I leave the decision whether the version should be "nightly" or "unstable" to the maintainer or whoever merges the PR.

LGTM. Tested pulseview with a logic analyzer and smuview with a multimeter.

Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python310Packages.sigrok</li>
    <li>python311Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1162

panicgh commented:
Commit 4b030f5a42c65eb3c36361c058c34846a36f8376 does not seem to belong into this PR.

vifino commented:
Hey @panicgh, thanks for noticing, fixed.

panicgh reviewed (approved):
LGTM. `collectd` fails because of an include file error unrelated to sigrok (see hydra).

Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>collectd</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python310Packages.sigrok</li>
    <li>python311Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1215

panicgh commented:
collectd recovered.

Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python310Packages.sigrok</li>
    <li>python311Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

ius reviewed (changes requested):
Built & tested - one minor comment regarding package naming.

Thanks for doing this, delighted to find someone beat me to it.

vifino commented on pkgs/applications/science/electronics/pulseview/default.nix:
```diff
@@ -1,45 +1,33 @@
 stdenv.mkDerivation rec {
   pname = "pulseview";
-  version = "0.4.2";
+  version = "nightly-2023-10-21";
```
Hi!

I've got the same suggestion [two months ago](https://github.com/NixOS/nixpkgs/pull/259629#discussion_r1349656840).

Since upstream preferred nightly, I chose nightly.
But if this is what stops this from getting merged, I'll change it to unstable.

ius replied:
> But if this is what stops this from getting merged, I'll change it to unstable.

Just a matter of preference I guess. In any case, I can't merge anyway.

panicgh replied:
I much prefer `unstable` over `nightly`. It may be that upstream calls them `nightly` because they rebuild them every day (or night), but here at nixpkgs we won't do this and instead will just bump the version every now and then (every couple of weeks I guess).

vifino replied:
Majority wins - I updated it to unstable vocabulary (and updated it to the latest commits available now)

panicgh commented on pkgs/development/tools/libsigrok/default.nix:
```diff
@@ -19,16 +20,17 @@
+  version = "unstable-2023-11-21";
 
-  src = fetchurl {
-    url = "https://sigrok.org/download/source/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "0g6fl684bpqm5p2z4j12c62m45j1dircznjina63w392ns81yd2d";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}";
+    rev = "201bd256a8aa85035423023914889fe9f5db52db";
+    hash = "sha256-A1gH07bmNeiFM/yTQ6fE75mrMhGHC+kps7zy6zI51E4=";
```
You have updated the version and the commit message, but not the source `rev` and `hash`, which are the same as for the 2023-10-18 version ([rev](https://sigrok.org/gitweb/?p=libsigrok.git;a=commit;h=201bd256a8aa85035423023914889fe9f5db52db)).

panicgh reviewed (approved):
Tested sigrok-cli and pulseview with a LA, tested smuview with a DMM.

Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python310Packages.sigrok</li>
    <li>python311Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

panicgh commented:
Still everything still builds fine. I use the changes in this PR for months now on 23.11 and had no issues so far.

Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python310Packages.sigrok</li>
    <li>python311Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1356

SuperSandro2000 commented on pkgs/development/tools/sigrok-firmware-fx2lafw/default.nix:
```diff
@@ -1,21 +1,23 @@
-    url = "https://sigrok.org/download/source/sigrok-firmware-fx2lafw/sigrok-firmware-fx2lafw-${version}.tar.gz";
-    sha256 = "sha256-o/RA1qhSpG4sXRmfwcjk2s0Aa8BODVV2KY7lXQVqzjs=";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}";
```
```suggestion
    url = "git://sigrok.org/sigrok-firmware-fx2lafw";
```
a change to pname likely will make this wrong

SuperSandro2000 commented on pkgs/development/tools/sigrok-cli/default.nix:
```diff
@@ -1,22 +1,23 @@
-    url = "https://sigrok.org/download/source/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "sha256-cdBEPzaJe/Vlcy3sIGgw2+oPJ4m2YBzxBTayhtEUCrg=";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}";
```
```suggestion
    url = "git://sigrok.org/sigrok-cli";
```

SuperSandro2000 commented on pkgs/development/tools/libsigrokdecode/default.nix:
```diff
@@ -1,19 +1,15 @@
-    url = "https://sigrok.org/download/source/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "1h1zi1kpsgf6j2z8j8hjpv1q7n49i3fhqjn8i178rka3cym18265";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}/";
```
```suggestion
    url = "git://sigrok.org/libsigrokdecode/";
```

SuperSandro2000 commented on pkgs/development/tools/libsigrok/default.nix:
```diff
@@ -19,16 +20,17 @@
-    url = "https://sigrok.org/download/source/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "0g6fl684bpqm5p2z4j12c62m45j1dircznjina63w392ns81yd2d";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}";
```
```suggestion
    url = "git://sigrok.org/libsigrok";
```

SuperSandro2000 commented on pkgs/applications/science/electronics/pulseview/default.nix:
```diff
@@ -1,45 +1,33 @@
-    url = "https://sigrok.org/download/source/pulseview/pulseview-${version}.tar.gz";
-    hash = "sha256-8EL3ej4bNb8wZmMw427Dj6uNJIw2k8N7fjXUAcO/q8s=";
+  src = fetchgit {
+    url = "git://sigrok.org/${pname}";
```
```suggestion
    url = "git://sigrok.org/pulseview";
```

SuperSandro2000 reviewed (commented):
substituting pname wherever possible is an anti pattern. I also have doubts that we should backport an update to unstable versions.

panicgh commented:
@vifino kind ping... It would be unfortunate to loose this work.

vifino commented:
Thanks for the ping @panicgh! Was at a good time. :)

Rebased, removed the substitution in the url (even though I don't necessarily agree) and updated to the latest commits.

panicgh reviewed (approved):
Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python311Packages.sigrok</li>
    <li>python312Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1404

OPNA2608 reviewed (changes requested):
`version` attributes should be `<latest version>-unstable-<YYYY-MM-DD>`, see [/pkgs/README.md#package-naming](https://github.com/NixOS/nixpkgs/blob/0e03f6a2d4d9eacdab1ff8d19b65c7950d580229/pkgs/README.md#package-naming).

And the new `smuview` should ideally be under the new `/pkgs/by-name`. To make that work, in addition to moving the file, you'll need to use `qt5` to get `wrapQtAppsHook`.

panicgh commented:
> `version` attributes should be `<latest version>-unstable-<YYYY-MM-DD>`, see [/pkgs/README.md#package-naming](https://github.com/NixOS/nixpkgs/blob/0e03f6a2d4d9eacdab1ff8d19b65c7950d580229/pkgs/README.md#package-naming).

While this is what the file says, it seems very unpopular in the repository as is and also the [nixpkgs doc contains](https://nixos.org/manual/nixpkgs/unstable/#ssec-dart-flutter) an example just using a "unstable-\<yyyy-mm-dd\>" version (the *only* example with unstable version!).

```console
$ git grep -l 'version.*=.*"[[:digit:]].*unstable' pkgs/*.nix  | wc -l
103
$ git grep -l 'version.*=.*"unstable' pkgs/*.nix |  wc -l
1518
```

I, therefore, favor finally merging these changes and let someone else cleanup the unstable versions nixpkgs-wide and in the documentation if it is regarded as relevant, or change the readme to reflect the widely used version name scheme.

OPNA2608 commented:
> it seems very unpopular in the repository
> I […] favor […] [letting] someone else cleanup the unstable versions nixpkgs-wide

Disagree.

1. The change to our guidelines was made with no real announcements afew months ago (#234201) so many people likely didn't know that it was eventually made, but the details of its formatting have been discussed for a [long](https://github.com/NixOS/rfcs/pull/107) [time](https://github.com/NixOS/rfcs/pull/147). It's starting to be used in [recent](https://github.com/NixOS/nixpkgs/commit/613e86b7627e154b2ed844ba8a3a7345533aba46) [man-made](https://github.com/NixOS/nixpkgs/commit/4d2152d75f59ef1f10f112b0ee6528d3732929d6) [inits/bumps](https://github.com/NixOS/nixpkgs/commit/e55adab288be297b7653f71ab224b90313d03013).
2. Other documentation is simply outdated when it doesn't use the new guideline. (But I *will* update the example you pointed out, for consistency's sake :wink:)
3. "It's comparatively unpopular" also applies to the new `/pkgs/by-name` format - `1092` instances of `/pkgs/by-name/fo/foobar/package.nix` pattern, vs the entire rest of Nixpkgs.
4. "I'll just let someone else clean this up" just adds more work and points-of-conflict for whoever "wants" (read: feels most annoyed by non-compliant code) to fix the entirety of Nixpkgs.

I see none of these points as good reasons for not fixing this here & now, before it comes more annoying to properly address this post-merge.

If the only leftover issue is that you don't want this to go unmerged for an eternity: Since I'm currently using sigrok stuff for university work, I'll gladly stay on the ball w/r/t reviews and merge this - once I'm happy with its guideline compliance.

panicgh commented:
I wouldn't mind fixing this PR if we weren't gambling with the author's time and patience, who is waiting for 4 months to get it merged. It was originally intended for 23.11 and created well due in time. Still the improvements are not yet merged (the upstream releases are from 2019-2021, massively outdated).
While I'd welcome a guideline compliant update from the author, I can honestly understand if they gave up on it. If I was them and had not given up, I would do only what is necessary to get over the compliance issues, and not rebase or bump it since that invokes another round of reviews and related delays. This could be done in a follow-up PR.

I could help to make this pkg comply to the latest guideline and bump it from time to time, but as long as this PR isn't merged I cannot do anything. I am happy about your offer to keep an eye on this pkg.

carlossless commented:
Hey, just bumping this for you @vifino. (as I am also interested in seeing this merged in)

If I had any say in the discussion, I'd vote for making the (seemingly trivial) changes to conform to the guidelines so that we could see this in master soon!

vifino commented:
I do not understand the purpose of insisting on dragging a version number around if it bears no significance anymore.

I understand that scheme when a specific unreleased "hot fix" commit was chosen and the project intends to cut another release in the future.
In that case, it is quite sane and useful.

But if the version is years behind and the project has no plans to release a new version, that number is meaningless.
When I see stuff like `0-unstable` I get infuriated.

Besides, this is a *should*, not a *must*. Why *must* this be enforced here?

---

Regardless of what I think on this issue, after more than 4 months of keeping this up to date and addressing feedback - including the change from `nightly-YYYY-MM-DD` to `unstable-YYYY-MM-DD` (against explicit upstreams wishes) - my willingness to put further effort into this is quite low.

I will update this PR one last time.

OPNA2608 reviewed (approved):
Result of `nixpkgs-review pr 259629` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>libsigrok</li>
    <li>libsigrokdecode</li>
    <li>pulseview</li>
    <li>python311Packages.sigrok</li>
    <li>python312Packages.sigrok</li>
    <li>sigrok-cli</li>
    <li>sigrok-firmware-fx2lafw</li>
    <li>smuview</li>
  </ul>
</details>

---

(also built everything on aarch64-linux, but can't run `nixpkgs-review` there)

Checked `pulseview` with the logic analyser at my uni's lab, and `sigrok-cli` gave the output I'm used to from Debian. LGTM.

vifino commented:
Thank you.
